### ISSUE TITLE
Image mean vs pixel mean substraction

### ISSUE THREAD

**Issue by willishf** [NVIDIA/DIGITS#1551]
Looks like I have run into the mean pixel vs mean image problem comparing results from DIGITS 5.0 interface to method used in example.py. It appears to have a significant impact and based on searching the forum can only find a discussion with no solution.

I have already trained model with mean image subtraction which was the default and it took 3 days to train the model. In testing the model using the Digits web interface classify many very happy with the results in that for the primary target that I want to identify minimal false positives in other test images and near perfect on the primary image. Working on an April 1st deadline for a cancer tumor classification challenge so don't have time to do anything significantly different at this point.

The images that I am submitting to example.py are already 256x256 so they don't need to be resized. The images I submitted to Caffe for training were also 256x256 so shouldn't have been resized related to the potential of a resizing problem.

I am passing the mean file on the command line from the model via --mean mean.binaryproto which is triggering if mean_file in the get_transformer(deploy_file, mean_file=None)

The comment clearly indicates #set mean pixel as the method. In looking through the code where I have no familiarity with the API I don't see an obvious path to doing a mean of the image where the assumption it should be a trivial change.

Is it possible that someone who is familiar with the API could look at the code and provide the changes that are required to do mean image subtraction instead of mean pixel subtraction?

The results I am getting with test images using the model that appears to be very good using the DIGITS web interface is not performing well via the example.py approach. I am already setup with a pipeline that has been tested using example.py for many many many images so hoping for a one or two line code update to improve results.

I also kicked off a version 2 of the model with 30 percent more images and to resolve some image export race conditions I had where the top half of the image would be blank because I was writing before the image was fully loaded.  This time I selected mean pixel for building the model in DIGITS 5.0. Gone through 6 epochs and have a set of test images that I am using. For the 6.6 million images using for training and 2.2 million images using for validation getting accuracy of 97%. However, when I submit a group of test images for one specific label in epoch 5 it mapped them all to the wrong label and in epoch 6 mapped them all wrong to an entirely different label. Not sure if this is related to using mean pixel as opposed to mean image for training where so far the results are not very good. 

If the default for the DIGITS 5.0 web interface is mean image is that because it is the optimal choice or simply an entry for the drop down. Don't want to spend the next 8 days hoping the models improve. Also don't understand why it is reporting 97% accuracy for the validation images. I can't imagine overfitting with 6.6 million training images and 2 .2 million validation images.

**Comment by Vincechen611** [comment]
I have got this same issue.
The result from mean piexl is worse than mean image subtraction.
I am also confuse why it is reporting high accuracy for the validation images during training.

**Comment by zhuozhongshuo** [comment]
@lukeyeager,  in  [binary-segmentation](https://github.com/NVIDIA/DIGITS/tree/master/examples/binary-segmentation) example, the introduction recommends us to use 'mean pixel' skill. However, I can not find the mean file (xxx.binaryproto) in that certain created dataset job. What's more, the [segmentation-model.prototxt](https://github.com/NVIDIA/DIGITS/blob/master/examples/binary-segmentation/segmentation-model.prototxt) does not include "mean_value" in data layer, which denotes mean pixel operation.
It is more amazing that when I use pixel mean in pycaffe, the output is extremely poor. Yet, when I feed the input image into the network directly, the output is closer to those results in digits.
I am really confused. Could you give me some some advices? Thank you.
